mycroft/skills/main.py

```diff
@@ -278,14 +280,16 @@ def _load_or_reload_skill(self, skill_folder):
             # removing listeners and stopping threads
             skill["instance"].shutdown()
 
-            # Remove two local references that are known
-            refs = sys.getrefcount(skill["instance"]) - 2
-            if refs > 0:
-                LOG.warning(
-                    "After shutdown of {} there are still "
-                    "{} references remaining. The skill "
-                    "won't be cleaned from memory."
-                    .format(skill['instance'].name, refs))
+            if DEBUG:
```

Comment:
Just to clarify, `gc.collect()` is normally automatically run, right? So all this does is make sure it's run right before checking references?

Reply:
Yes that's precisely it.